employee-rostering-frontend/src/ui/components/DataTable.tsx

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import {
+  Table,
+  TableHeader,
+  TableBody,
+  headerCol,
+  ICell,
+  IRow,
+  IAction
+} from '@patternfly/react-table';
+
+export interface DataTableProps<T> {
+  title: string;
+  columnTitles: string[];
+  tableData: T[];
+  rowDataToRow: (rowData : T) => IRow;
+}
+
+interface DataTableState<T> {
+  currentFilter: (_: T) => boolean;
+}
+
+export abstract class DataTable<T, Props extends DataTableProps<T>> extends React.Component<Props, DataTableState<T>> {
+  columns: ICell[];
+  actions: IAction[];
+
+  constructor(props: Props) {
+    super(props);
+    this.columns = props.columnTitles.map(t => { return { title: t, cellTransforms: [headerCol], props: {} }; });
```

Comment:
You "infer" `columns` from `props.columnTitles` in the constructor. This is a one time operation and you effectively disconnect the result of `render()` from `props`. This way, when props are changed from the outside and React will try to re-render this component, it won't reflect the change in props, because you're using this "cached" value.